[FIX] sms: sending sms required field issue

niyasraphy opened the pull request:
**Description of the issue/feature this PR addresses:**
Impacted versions:
16 CE and EE

**Steps to reproduce:**

From Calendar menu, switch to list view
Select multiple meeting records
Click on Send SMS button in the header of the tree
Click Put In Queue or Send SMS without entering the message content
Error will be raised as the message content(body) field is required at model level

**Current behavior before PR:**
Exception is raised without sending SMS.

Desired behavior after PR is merged:
Send SMS or add SMS to queue without exceptions.



---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/103718).

niyasraphy commented:
Currently the required attribute from the model level is removed, so that the error wont be raised and already there is some conditions in view level to make it required.
Is this better solution or should it be done in better way ? like, always required in view level or validation on clicking the buttons?
Your thoughts !

Yenthe666 commented:
Technically that is a model change which is not allowed in a stable version but let's see what Odoo says :)

abdounasser202 commented on addons/sms/wizard/sms_composer.py:
```diff
@@ -65,7 +65,7 @@ def default_get(self, fields):
     template_id = fields.Many2one('sms.template', string='Use Template', domain="[('model', '=', res_model)]")
     body = fields.Text(
         'Message', compute='_compute_body',
-        precompute=True, readonly=False, store=True, required=True)
```
I think this field should be required, it doesn't make sense to send an empty SMS

niyasraphy replied:
@abdounasser202 thanks for the feedback, updated

niyasraphy commented:
@abdounasser202 thanks for the feedback, 
modified the pr to keep the required attribute in the model level and removed the conditional required attribute given in view level, so that when we click the send button it will ask us to enter the message/body if the required field is empty.
Is this okay ?

abdounasser202 commented:
Fine for me

abdounasser202 commented:
@niyasraphy

tde-banana-odoo commented:
Hello,

Thanks for the PR, seems legit at first sight. Addition of this strange required attribute was made at odoo/odoo@08a5bf761fffcc05b633149de4d9021251d4f587 . Indeed I would keep the standard required one at model level. @kba-odoo do you remember why you added some required attribute on the view itself ? As it makes no sense to try to send void SMS...

Cheers !

phwa-odoo commented:
@tde-banana-odoo Probably a leftover from this conversation https://github.com/odoo/odoo/pull/83745#discussion_r817623688

niyasraphy commented:
@phwa-odoo if this is the case, the given condition seems to be wrong and not working as expected,
putting an opposite condition in the current required attribute will resolve the case in this comment:  https://github.com/odoo/odoo/pull/83745#discussion_r817623688

Should i modify the PR with that condition or its fine with this ?
@tde-banana-odoo

niyasraphy commented:
ping @phwa-odoo @tde-banana-odoo

abdounasser202 reviewed (approved):
Now looks good and make sense

tde-banana-odoo commented:
@robodoo r+

tde-banana-odoo commented:
@robodoo r-

tde-banana-odoo commented:
Hello @niyasraphy ,

Could you elaborate commit message ? You should explain what is the issue and why we do that change :)

Cheers !

niyasraphy commented:
@tde-banana-odoo updated commit message and pr title

tde-banana-odoo commented:
@robodoo r+
